### PR TITLE
Show fees

### LINKED ISSUE #2238: Show fees per transaction
This is a sub-issue for fee calculation and displaying #1721

AC:
- Display calculated fee information before signing transactions

### PR DESCRIPTION
**Pull request by fedekunze**
Closes #2238

**Description:**

Screen Shot 2019-03-25 at 1 49 18 PM
Screen Shot 2019-03-25 at 1 49 35 PM

Thank you! 🚀

---

For contributor:

- [ ] Added entries in `PENDING.md` with issue # and GitHub username
- [ ] Reviewed `Files changed` in the github PR explorer
- [ ] Attach screenshots of the UI components on the PR description (if applicable)
- [ ] Scope of work approved for big PRs

For reviewer:

- [ ] Manually tested the changes on the UI

### CODE REVIEW

**Comment by fedekunze** [comment]
@jbibla can you style the table invoice when you have time ?

File: app/src/renderer/components/common/ActionModal.vue [hunk 1]
```diff
@@ -181,20 +224,44 @@ export default {
...
+      default: `0`
     }
   },
   data: () => ({
     step: defaultStep,
     signMethod: null,
     password: null,
     sending: false,
+    gasEstimate: null,
+    gasPrice: (0.025 * 1e-6).toFixed(9), // recomended default: 0.025 uatom per gas
```
Comment (faboweb) [inline app/src/renderer/components/common/ActionModal.vue]: can we extract this constant into `config.js`?

File: app/src/renderer/components/common/TableInvoice.vue [hunk 2]
```diff
@@ -0,0 +1,73 @@
...
+      type: Number,
+      required: true
+    },
+    gasEstimate: {
+      type: Number,
+      required: true
+    },
+    gasPrice: {
+      type: Number,
+      default: 2.5e-8 // 0.025 uatoms
```
Comment (faboweb) [inline app/src/renderer/components/common/TableInvoice.vue]: same: would be nice to extract

File: app/src/renderer/components/common/TableInvoice.vue [hunk 3]
```diff
@@ -0,0 +1,73 @@
...
+    }
+  },
+  data: () => ({
+    full,
+    info: `Estimated fees required to process the transaction, based on simulation. Cosmos Wallet doesn't take any fees.`
+  }),
+  computed: {
+    estimatedFee() {
+      const gasAdjustment = 1.5
```
Comment (faboweb) [inline app/src/renderer/components/common/TableInvoice.vue]: same

File: app/src/renderer/components/common/TableInvoice.vue (hunk 2 above)
Comment (fedekunze) [reply app/src/renderer/components/common/TableInvoice.vue]: updated

File: app/src/renderer/components/common/TableInvoice.vue (hunk 3 above)
Comment (fedekunze) [reply app/src/renderer/components/common/TableInvoice.vue]: updated

File: app/src/renderer/components/common/ActionModal.vue (hunk 1 above)
Comment (fedekunze) [reply app/src/renderer/components/common/ActionModal.vue]: updated

File: app/src/config.js [hunk 4]
```diff
@@ -21,7 +21,8 @@ export default {
   sentry_dsn: `https://4dee9f70a7d94cc0959a265c45902d84:cbf160384aab4cdeafbe9a08dee3b961@sentry.io/288169`,
   node_halted_timeout: 120000,
   block_timeout: 10000,
-  default_gas: 500000,
+  default_gas_price: 2.5e-8, // recomended from Cosmos Docs
```
Comment (faboweb) [inline app/src/config.js]: nice

File: app/src/renderer/vuex/modules/governance/deposits.js [hunk 5]
```diff
@@ -1,3 +1,4 @@
+
```
Comment (faboweb) [inline app/src/renderer/vuex/modules/governance/deposits.js]:
```suggestion
```

File: app/src/renderer/vuex/modules/delegation.js [hunk 6]
```diff
@@ -182,12 +207,31 @@ export default ({ node }) => {
...
+    },
     async submitUnbondingDelegation(
       {
         rootState: { session },
         dispatch
       },
-      { validator, amount, password, submitType }
+      { validator, amount, gas, gas_prices, password, submitType }
     ) {
       // TODO: change to 10 when available https://github.com/cosmos/cosmos-sdk/issues/2317
```
Comment (faboweb) [inline app/src/renderer/vuex/modules/delegation.js]:
```suggestion
```

File: app/src/renderer/vuex/modules/delegation.js [hunk 7]
```diff
@@ -182,12 +207,31 @@ export default ({ node }) => {
...
       dispatch(`updateDelegates`)
     },
+    async simulateUnbondingDelegation(
+      {
+        rootState: { session },
+        dispatch
+      },
+      { validator, amount }
+    ) {
+      // TODO: change to 10 when available https://github.com/cosmos/cosmos-sdk/issues/2317
```
Comment (faboweb) [inline app/src/renderer/vuex/modules/delegation.js]:
```suggestion
```

File: app/src/renderer/vuex/modules/send.js [hunk 8]
```diff
@@ -62,94 +60,69 @@ export default ({ node }) => {
...
-      // TODO: remove once fees are displayed on action modals
-      console.log(
-        `- Estimated Gas: ${simulationRes.gas_estimate}\n` +
-        `- Adjusted Gas: ${adjustedGas}\n` +
-        `- Estimated Fees (low): ${estimatedFeesLow} ${gasPrices[0].denom}s\n` +
-        `- Estimated Fees (high): ${estimatedFeesHigh} ${gasPrices[0].denom}s`
-      )
-      args.base_req.simulate = false
-      args.base_req.gas = simulationRes.gas_estimate // adjusted on the SDK
+      const properties = [`submitType`, `type`, `to`, `pathParameter`, `gas`, `gas_prices`, `simulate`]
```
Comment (faboweb) [inline app/src/renderer/vuex/modules/send.js]: code smell

File: test/unit/specs/store/governance/proposals.spec.js [hunk 9]
```diff
@@ -135,15 +135,15 @@ describe(`Module: Proposals`, () => {
 
   describe(`Fetch a single proposal`, () => {
     it(`when the request is successful`, async () => {
-      proposalsModuleInstance = proposalsModule({
+      module = proposalsModule({
```
Comment (faboweb) [inline test/unit/specs/store/governance/proposals.spec.js]: module is a reserved keyword and shouldn't be used

File: app/src/renderer/vuex/modules/send.js (hunk 8 above)
Comment (fedekunze) [reply app/src/renderer/vuex/modules/send.js]: ?

Comment (faboweb) [reply app/src/renderer/vuex/modules/send.js]: manipulating this `args` object is not how we should do things. as it is dangerous as we might forget properties.

**Pull request merged**